core/drivers/crypto/caam/crypto.mk

```diff
@@ -132,6 +132,12 @@ CFG_NXP_CAAM_RSA_KEY_FORMAT ?= 3
 # Disable device tree status of the secure job ring
 CFG_CAAM_JR_DISABLE_NODE ?= y
 
+# Define the default CAAM private key encryption generation and the bignum
+# maximum size needed.
+# CAAM_KEY_PLAIN_TEXT    -> 4096 bits
+# CAAM_KEY_BLACK_ECB|CCM -> 4156 bits
+CFG_CORE_BIGNUM_MAX_BITS = 4156
```

Comment:
`?=`

Are 4096 and 4156 the only valid values? If so it may be good to check it with `ifneq($(CFG_CORE_BIGNUM_MAX_BITS),...) ... $(error ...) ... endif`

Reply:
4096 and 4156 are the minimum values required for 4096 RSA keypairs and black keypairs. You could still be able to 
decrease these values if you don't want to generate 4096 keypairs.

Here I check if `CFG_CORE_BIGNUM_MAX_BITS` can store a 4096 bits RSA black keypair
https://github.com/OP-TEE/optee_os/pull/5990/files#diff-2af92fa52b0087bf848ecd8b9b9de69c7330aa1b63cb31c3388387e1dcdb5bd7R723-R727

Reply:
4096 and 4156 are the minumum value s

Here I check if `CFG_CORE_BIGNUM_MAX_BITS` can store a 4096 bits RSA black keypair
https://github.com/OP-TEE/optee_os/pull/5990/files#diff-2af92fa52b0087bf848ecd8b9b9de69c7330aa1b63cb31c3388387e1dcdb5bd7R723-R727

Reply:
I have to set it like this `=` instead of `?=` otherwise it won't override the default value.

Reply:
But that's not allowed (see [here](https://optee.readthedocs.io/en/latest/building/gits/optee_os.html#there-are-only-two-ways) and a in the paragraphs above for details). It's either `?=` or `$(call force,CFG_CORE_BIGNUM_MAX_BITS,4156)`. In this case, if a default value is set somewhere else, you should try to set it earlier so that it becomes the actual default.

Reply:
Fix suggestion here : ba8ef3776